internal/cmd/controller/target/builder.go

```diff
@@ -122,24 +123,30 @@ func (m *Manager) Targets(ctx context.Context, bundle *fleet.Bundle, manifestID
 
 	byNamespace := map[string]*fleet.BundleDeployment{}
 	for _, bd := range bundleDeployments.Items {
+		bd := bd.DeepCopy()
```

Comment:
`bundleDeployments.Items` may be the result from reading from the cache. Since we perform changes in this object, it's safer to work with a copy of it.